tests/utils/data/test_utils.py

```
@@ -230,6 +231,65 @@ def test_dataset_without_input_ids_column(self):
         self.assertEqual(len(result), len(dataset))
         self.assertListEqual(list(result.column_names), ["chosen", "rejected"])
 
+
+class TestLocalFileSplitHandling(unittest.TestCase):
+    """Regression tests for local-file dataset split handling."""
+
+    @patch("axolotl.utils.data.shared.load_dataset")
+    def test_local_jsonl_preserves_split_slice(self, mock_load_dataset):
+        dataset_config = DictDefault(
+            {
+                "path": "/tmp/sample.jsonl",
```

Ruff (0.15.11)

[error] Probable insecure usage of temporary file or directory: "/tmp/sample.jsonl"

(S108)